test/spec/proxySpec.js

```diff
@@ -1148,6 +1146,86 @@ describe('proxy', function() {
       // }, 100);
     });
   });
+  describe('Already proxified values', function() {
+    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
```

Comment:
>Could you give an example of such? 
```js
var originalObject = {foo: 'bar'}

// assuming JSONPatcherProxy doesn't deep clone
var jsonPatcherProxy = new JSONPatcherProxy(originalObject);
var proxy = jsonPatcherProxy.observe(true);

console.log(proxy.foo) // --> bar
originalObject.bees = 'fly'
console.log(proxy.bees) // --> fly (this is the issue, setting a value to the original object allows it inside the proxy)

originalObject.beesObject = {bees: 'fly'}; 
console.log(proxy.beesObject) // --> unproxified {bees: 'fly'} 
```

And the second argument is more important. Proxifiying an object proxifies its children in place.